framework/src/transfers/MultiAppNearestNodeTransfer.C

```diff
@@ -345,38 +344,38 @@ MultiAppNearestNodeTransfer::execute()
           _from_meshes[i], local_entities[i], local_comps[i], is_to_nodal, is_constant);
     }
 
-    if (_fixed_meshes)
-    {
-      _cached_froms.resize(n_processors());
-      _cached_dof_ids.resize(n_processors());
-    }
+    // Quadrature points I will receive from remote processors
+    std::map<processor_id_type, std::vector<Point>> incoming_qps;
+    auto qps_action_functor = [&incoming_qps](processor_id_type pid,
+                                              const std::vector<Point> & qps) {
+      // Quadrature points from processor 'pid'
+      auto & incoming_qps_from_pid = incoming_qps[pid];
+      // Store data for late use
+      std::copy(qps.begin(), qps.end(), std::back_inserter(incoming_qps_from_pid));
+    };
 
-    for (processor_id_type i_proc = 0; i_proc < n_processors(); i_proc++)
+    Parallel::push_parallel_vector_data(comm(), outgoing_qps, qps_action_functor);
+
+    for (auto & qps : incoming_qps)
     {
-      // We either use our own outgoing_qps or receive them from
-      // another processor.
-      std::vector<Point> incoming_qps;
-      if (i_proc == processor_id())
-        incoming_qps = outgoing_qps[i_proc];
-      else
-        _communicator.receive(i_proc, incoming_qps);
+      processor_id_type i_proc = qps.first;
```

Comment:
Now that you're moving a map, you should switch `i_proc -> pid` for everything in this loop. I think that you can

```suggestion
      const processor_id_type pid = qps.first;
```